pkg/scheduler/util/utils.go

```diff
@@ -146,3 +146,227 @@ func MoreImportantPod(pod1, pod2 interface{}) bool {
+	expression := metav1.LabelSelectorRequirement{
+		Key:      key,
+		Operator: metav1.LabelSelectorOpNotIn,
+		Values:   vals,
+	}
+	s.MatchExpressions = append(s.MatchExpressions, expression)
+	return s
+}
+
+// Exists injects a matchExpression (with an operator Exists) to the inner labelSelector
+func (s *LabelSelectorWrapper) Exists(k string) *LabelSelectorWrapper {
+	expression := metav1.LabelSelectorRequirement{
+		Key:      k,
+		Operator: metav1.LabelSelectorOpExists,
+	}
+	s.MatchExpressions = append(s.MatchExpressions, expression)
+	return s
+}
+
+// NotExist injects a matchExpression (with an operator NotExist) to the inner labelSelector
+func (s *LabelSelectorWrapper) NotExist(k string) *LabelSelectorWrapper {
+	expression := metav1.LabelSelectorRequirement{
+		Key:      k,
+		Operator: metav1.LabelSelectorOpDoesNotExist,
+	}
+	s.MatchExpressions = append(s.MatchExpressions, expression)
+	return s
+}
+
+// Obj returns the inner LabelSelector
+func (s *LabelSelectorWrapper) Obj() *metav1.LabelSelector {
+	return &s.LabelSelector
+}
+
+// PodWrapper wraps a Pod inside.
+type PodWrapper struct{ v1.Pod }
+
+// MakePod creates a Pod wrapper
+func MakePod() *PodWrapper {
+	return &PodWrapper{v1.Pod{}}
+}
+
+// Obj returns the inner Pod
+func (p *PodWrapper) Obj() *v1.Pod {
+	return &p.Pod
+}
+
+// Name sets `s` as the name of the inner pod
+func (p *PodWrapper) Name(s string) *PodWrapper {
+	p.SetName(s)
+	return p
+}
+
+// Namespace sets `s` as the namespace of the inner pod
+func (p *PodWrapper) Namespace(s string) *PodWrapper {
+	p.SetNamespace(s)
+	return p
+}
+
+// Node sets `s` as the nodeName of the inner pod
+func (p *PodWrapper) Node(s string) *PodWrapper {
+	p.Spec.NodeName = s
+	return p
+}
+
+// NodeSelector sets `m` as the nodeSelector of the inner pod
+func (p *PodWrapper) NodeSelector(m map[string]string) *PodWrapper {
+	p.Spec.NodeSelector = m
+	return p
+}
+
+// NodeAffinityIn creates a HARD node affinity (with the operator In)
+// and injects into the innner pod
+func (p *PodWrapper) NodeAffinityIn(key string, vals []string) *PodWrapper {
+	if p.Spec.Affinity == nil {
+		p.Spec.Affinity = &v1.Affinity{}
+	}
+	if p.Spec.Affinity.NodeAffinity == nil {
+		p.Spec.Affinity.NodeAffinity = &v1.NodeAffinity{}
+	}
+	nodeSelector := MakeNodeSelector().In(key, vals).Obj()
+	p.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = nodeSelector
+	return p
+}
+
+// NodeAffinityNotIn creates a HARD node affinity (with the operator NotIn)
+// and injects into the innner pod
+func (p *PodWrapper) NodeAffinityNotIn(key string, vals []string) *PodWrapper {
+	if p.Spec.Affinity == nil {
+		p.Spec.Affinity = &v1.Affinity{}
+	}
+	if p.Spec.Affinity.NodeAffinity == nil {
+		p.Spec.Affinity.NodeAffinity = &v1.NodeAffinity{}
+	}
+	nodeSelector := MakeNodeSelector().NotIn(key, vals).Obj()
+	p.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = nodeSelector
```

Comment:
why do we care about the affinity fields in this PR ?I would try to keep this feature completely separate from the affinity feature